moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -416,17 +423,39 @@ class KinematicsBase
     if (tip_frames.size() == 1)
     {
       return initialize(robot_description,
-        group_name,
-        base_frame,
-        tip_frames[0],
-        search_discretization);
+                        group_name,
+                        base_frame,
+                        tip_frames[0],
+                        search_discretization);
     }
 
     logError("moveit.kinematics_base: This kinematic solver does not support initialization with more than one tip frames");
     return false;
   }
 
   /**
+   * @brief  Initialization function for the kinematics, for use with kinematic chain IK solvers
+   * @param robot_model - allow the URDF to be loaded much quicker by passing in a pre-parsed model of the robot
+   * @param group_name The group for which this solver is being configured
+   * @param base_frame The base frame in which all input poses are expected.
+   * This may (or may not) be the root frame of the chain that the solver operates on
+   * @param tip_frame The tip of the chain
+   * @param search_discretization The discretization of the search when the solver steps through the redundancy
+   * @return True if initialization was successful, false otherwise
+   */
+  virtual bool initialize(const moveit::core::RobotModel* robot_model,
```

Comment:
I think it is dangerous to pass in a pure pointer here. Previously the kinematics plugin created and owned the RobotModel. If we pass in a pointer, it should be a shared_ptr to ensure existence of the RobotModel for the whole lifetime of the plugin.